code/game/objects/effects/decals/cleanable/misc.dm

```diff
@@ -89,10 +89,12 @@
 	addtimer(CALLBACK(src, .proc/Decay), 24 SECONDS)
 
 /obj/effect/decal/cleanable/greenglow/proc/Decay()
+	var/datum/component/radioactive/R = src.GetComponent(/datum/component/radioactive)
 	name = "dried goo"
 	light_power = 0
 	light_range = 0
 	update_light()
+	R.RemoveComponent()
```

Comment:
Move this up below where you fetch the component, also check if an R was actually returned